pkg/runtime/unstructured_test.go

```diff
@@ -132,6 +122,240 @@ func TestDecode(t *testing.T) {
+			"metadata": map[string]interface{}{
+				"name":              "test_name",
+				"namespace":         "test_namespace",
+				"generateName":      "test_generateName",
+				"uid":               "test_uid",
+				"resourceVersion":   "test_resourceVersion",
+				"selfLink":          "test_selfLink",
+				"creationTimestamp": "2009-11-10T23:00:00Z",
+				"deletionTimestamp": "2010-11-10T23:00:00Z",
+				"labels": map[string]interface{}{
+					"test_label": "test_value",
+				},
+				"annotations": map[string]interface{}{
+					"test_annotation": "test_value",
+				},
+				"ownerReferences": []map[string]interface{}{
+					{
+						"uid": "1",
+					},
+					{
+						"uid": "2",
+					},
+				},
+			},
+		},
+	}
+
+	if got, want := unstruct.GetAPIVersion(), "test_version"; got != want {
+		t.Errorf("GetAPIVersions() = %s, want %s", got, want)
+	}
+
+	if got, want := unstruct.GetKind(), "test_kind"; got != want {
+		t.Errorf("GetKind() = %s, want %s", got, want)
+	}
+
+	if got, want := unstruct.GetNamespace(), "test_namespace"; got != want {
+		t.Errorf("GetNamespace() = %s, want %s", got, want)
+	}
+
+	if got, want := unstruct.GetName(), "test_name"; got != want {
+		t.Errorf("GetName() = %s, want %s", got, want)
+	}
+
+	if got, want := unstruct.GetGenerateName(), "test_generateName"; got != want {
+		t.Errorf("GetGenerateName() = %s, want %s", got, want)
+	}
+
+	if got, want := unstruct.GetUID(), types.UID("test_uid"); got != want {
+		t.Errorf("GetUID() = %s, want %s", got, want)
+	}
+
+	if got, want := unstruct.GetResourceVersion(), "test_resourceVersion"; got != want {
+		t.Errorf("GetResourceVersion() = %s, want %s", got, want)
+	}
+
+	if got, want := unstruct.GetSelfLink(), "test_selfLink"; got != want {
+		t.Errorf("GetSelfLink() = %s, want %s", got, want)
+	}
+
+	if got, want := unstruct.GetCreationTimestamp(), unversioned.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC); !got.Equal(want) {
+		t.Errorf("GetCreationTimestamp() = %s, want %s", got, want)
+	}
+
+	if got, want := unstruct.GetDeletionTimestamp(), unversioned.Date(2010, time.November, 10, 23, 0, 0, 0, time.UTC); got == nil || !got.Equal(want) {
+		t.Errorf("GetDeletionTimestamp() = %s, want %s", got, want)
+	}
+
+	if got, want := unstruct.GetLabels(), map[string]string{"test_label": "test_value"}; !reflect.DeepEqual(got, want) {
+		t.Errorf("GetLabels() = %s, want %s", got, want)
+	}
+
+	if got, want := unstruct.GetAnnotations(), map[string]string{"test_annotation": "test_value"}; !reflect.DeepEqual(got, want) {
+		t.Errorf("GetAnnotations() = %s, want %s", got, want)
+	}
+
+	if got, want := unstruct.OwnerReferencesCount(), 2; !reflect.DeepEqual(got, want) {
+		t.Errorf("OwnerReferencesCount() = %d, want %d", got, want)
+	}
+
+	ref, _ := unstruct.GetOwnerReference(1)
+	if got, want := string(ref.UID), "2"; got != want {
+		t.Errorf("GetOwnerReference().UID=%s, want %s", got, want)
+	}
+}
+
+func TestUnstructuredOwnerReferenceSetters(t *testing.T) {
```

Comment:
Please roll this test into the setter test below.

Specifically I want to see all this work starting with a plain `&Unstructured{}`. I don't think you have a way to generically add owner's references when none exist.